ui/src/main/webapp/app/controller.js

```diff
@@ -0,0 +1,461 @@
+/*
```

Comment:
Would probably better to split this monolithic controller down a bit. Perhaps use angular services to encapsulate some of the logic and inject the service into the controller. 
Maybe even split the controller into multiple controllers that perform discrete functions